### DISCUSSION TITLE
Unable to connect srsUE to srsRAN gNB

### DISCUSSION THREAD

**Discussion by jonebert** [srsran/srsRAN_Project#691] [General Help]
Hi all,
I'm currently trying to setup srsRAN by following this [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#).
Sadly srsUE won't connect to the gNB and fails with the error message `Scheduling request failed: releasing RRC connection...`.
From the pcap's I can see that UE sends the "RRC Setup Complete" message but the gNB does not receive it.
I could only find this [issue](https://github.com/srsran/srsRAN_4G/issues/805) in the srsRAN_4G repo, which reported the same error but adjusting the `time_adv_samples` parameter did not fix this issue for me (I checked all values from 0 to 350 in 5 steps).
I use an USRP B200mini for the UE and an USRP B200mini-i for the gNB, but I also tried two B210 which also failed with the same error.
Here are my configurations, logs and pcaps.
[logs_and_configs.tar.gz](https://github.com/user-attachments/files/15986601/logs_and_configs.tar.gz)

Thank you in advance for any help!

**Comment by pgawlowicz** [comment]
could you try to increase the rx_gain in the srsgnb config?

**Reply by jonebert** [reply]
Thanks for the quick response. I tried it with rx_gain values from 40 to 75 but every time I got the same error.

**Reply by pgawlowicz** [reply]
could you post logs? also how is the setup connected? over the air or rf cables?

**Reply by jonebert** [reply]
The setup is over the air.
Here are the logs for a rx_gain of 75:
[gnb.log](https://github.com/user-attachments/files/15986939/gnb.log)
[ue.log](https://github.com/user-attachments/files/15986941/ue.log)

**Comment by pgawlowicz** [comment]
PDSCH received by UE has a good SNR:
```
2024-06-26T11:01:17.055298 [PHY0-NR] [I] [ 2081] PDSCH: cc=0 pid=0 si-rnti=0xffff prb=(1,8) symb=(2,13) CW0: mod=QPSK tbs=80 R=0.380 rv=0 CRC=OK iter=1.0 evm=0.21 t_us=193 epre=+5.2 snr=+17.6 cfo=+553.4 delay=+0.0  ack_tti_tx=2082
```
But the problem is with PUSCH at gnb:
```
PUSCH: rnti=0x460c h_id=0 prb=[3, 6) symb=[0, 14) mod=QPSK rv=0 tbs=11 crc=KO iter=6.0 sinr=-19.4dB t=211.4us uci_t=0.0us ret_t=0.0us
```

So the downlink from GNB to UE is good, but there is an issue with uplink. you need to tune gnb rx_gain and srsue tx_gain.

**Reply by jonebert** [reply]
I now tried gain values from 10 to 80 for gnb rx_gain and srsue tx_gain in all possible combinations but I'm still getting the same error message. Could there be any other reason for the bad uplink connection?

**Reply by pgawlowicz** [reply]
what value do you use for as `time_adv_nsamples` ?

**Reply by jonebert** [reply]
I'm currently not setting it manually so it is set to auto I assume.

**Reply by pgawlowicz** [reply]
could you try with 300? also you do not need to test all combinations, just a few is enough, and always try to increase the SNR at gnb,
you can display PUSCH stats at gnb at runtime with : 
```
tail -f /tmp/gnb.log | awk '/PUSCH:.*OK/ {print "\033[32m" $0 "\033[0m"; next} /PUSCH:.*KO/ {print "\033[31m" $0 "\033[0m"; next}'
```

**Reply by jonebert** [reply]
Thanks for the help! 

I retried it with `time_adv_nsamples` set to 300 and the RRC Connection was succesfully established, but now the gNB does not get the Authentication Response from the UE and is then released because the max number of retransmission is reached. Also the SNR of the uplink decreases drastically after the first few packages.
Here are the logs: 
[logs.tar.gz](https://github.com/user-attachments/files/15988984/logs.tar.gz)

**Reply by pgawlowicz** [reply]
could you repeat with debug logging mode at gnb?

**Reply by jonebert** [reply]
Sure here is the gNB log
[gnb.log](https://github.com/user-attachments/files/15989141/gnb.log)

**Reply by pgawlowicz** [reply]
It seems that CFO of the PUSCH is very high and fluctuates a lot, for example:
```
  cfo=-923.6Hz
  cfo=+1592.2Hz
  cfo=+1882.3Hz
  cfo=+1652.2Hz
  cfo=+1264.5Hz
```

Probably, you will need to use better clocks for your b200s.

**Reply by pgawlowicz** [reply]
@jonebert any update on this issue?

**Reply by jonebert** [reply]
Hi, yes the issue was resolved when I switched one of the minis with another one. It seems that the one I was using is faulty.
